### PR TITLE
Reactjs Responsive Web Design in One File Using a Custom Hook

### PR DESCRIPTION
**Pull request by unholydisaster**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+Make sure that you go to the correct project folder on your command prompt terminal and type this command.
+```cmd 
+//this takes you to the folder
+cd <project name>
+
+```
+install the styled-components
+
+```cmd
```
Comment (srishilesh) [inline content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md:31]: bash

File: content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+#### Topics:Reactjs.
+
+![Reactjs](/engineering-education/content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/hero.png "by the end, you will be able to do this")
+
+### IMPORTANT NOTE: The code and examples have been tested and work with the latest versions of React and styled-components.
+
+I recently fully converted my blog app with styled-components and am loving the number of complex functions one can add, from state management with props to implementing conditions on the CSS part.
+
+At first, I was using raw media queries for responsive design, the code took more than '1000' lines but what I am about to show you solves this problem.
+Let's get into the code guys.
```
Comment (srishilesh) [inline content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md:14]: Needs improvement. See my comment.

File: content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                  <>
+                      <div>
+                         <H1>this is a demo page</H1>
+                     </div>
+                  </>
+                  )
+     }
+}
+```
+
```
Comment (srishilesh) [inline content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md:119]: Why are we learning this? Explain them

File: content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+         return(
+                  <>
+                      <div>
+                         <h1>this is a demo page</h1>
+                     </div>
+                  </>
+                  )
+     }
+}
+```
```
Comment (srishilesh) [inline content/articles/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/index.md:86]: No much difference from previous snippet. Just a single line add. Avoid rewriting whole code snippets.

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @unholydisaster ,

Welcome to EngEd program!

On a quick glance at your PR, I find some major revisions that are to be done before one can start the review.

Here are few of my suggestions:
- Avoid using "I", instead prefer usage of "we", "us", "let's", and so on.
- The introduction must talk about the objectives of this tutorial, the problem faced, and solutions that you are going to provide. What are we learning? Why are we learning? How this article would help?
- Add Table of contents and pre-requisities sections
- I couldn't find any explanation of code snippets.
- State the objectives of the code tutorial
- Add necessary images that helps readers follow through the article
- `/Reactjs-Responsive-Web-Design-in-One-File-Using-a-Custom-Hook/` Make this in lower case

I would highly recommend to take a look at other approved PRs/published articles to see how other authors have structured and formatted their flow of the article. Your article needs some major improvements.

Moreover, your article is very low on content. Please add new topics to enhance it.

We will be closing this PR since it requires very major revisions. You may work on the changes in your forked branch.

Ping me in the new PR. once you have addressed all the changes.

Let me know in case of any questions.

Thanks.

**Comment by unholydisaster** [comment]
Hey, reactjs responsive web design in one file using a custom hook #7099

**Comment by unholydisaster** [comment]
Hey, thanks alot for checking my article and I appreciate the suggestions
that you have provided.

**Comment by unholydisaster** [comment]
I have created a new pull request, with the changes.
 Reactjs Responsive Web Design in One File Using a Custom Hook#7099

**Pull request closed without merging**